core/commands/add.go

```diff
@@ -69,26 +67,29 @@ You can now refer to the added file in a gateway, like so:
 	},
 	Options: []cmds.Option{
 		cmds.OptionRecursivePath, // a builtin option that allows recursive paths (-r, --recursive)
-		cmds.BoolOption(quietOptionName, "q", "Write minimal output.").Default(false),
-		cmds.BoolOption(silentOptionName, "Write no output.").Default(false),
+		cmds.BoolOption(quietOptionName, "q", "Write minimal output."),
+		cmds.BoolOption(silentOptionName, "Write no output."),
 		cmds.BoolOption(progressOptionName, "p", "Stream progress data."),
-		cmds.BoolOption(trickleOptionName, "t", "Use trickle-dag format for dag generation.").Default(false),
-		cmds.BoolOption(onlyHashOptionName, "n", "Only chunk and hash - do not write to disk.").Default(false),
-		cmds.BoolOption(wrapOptionName, "w", "Wrap files with a directory object.").Default(false),
-		cmds.BoolOption(hiddenOptionName, "H", "Include files that are hidden. Only takes effect on recursive add.").Default(false),
+		cmds.BoolOption(trickleOptionName, "t", "Use trickle-dag format for dag generation."),
+		cmds.BoolOption(onlyHashOptionName, "n", "Only chunk and hash - do not write to disk."),
+		cmds.BoolOption(wrapOptionName, "w", "Wrap files with a directory object."),
+		cmds.BoolOption(hiddenOptionName, "H", "Include files that are hidden. Only takes effect on recursive add."),
 		cmds.StringOption(chunkerOptionName, "s", "Chunking algorithm to use."),
-		cmds.BoolOption(pinOptionName, "Pin this object when adding.").Default(true),
+		cmds.BoolOption(pinOptionName, "Pin this object when adding.  Default: true."),
 	},
 	PreRun: func(req cmds.Request) error {
 		if quiet, _, _ := req.Option(quietOptionName).Bool(); quiet {
 			return nil
 		}
 
-		_, found, _ := req.Option(progressOptionName).Bool()
+		// ipfs cli progress bar defaults to true
+		progress, found, _ := req.Option(progressOptionName).Bool()
 		if !found {
-			req.SetOption(progressOptionName, true)
+			progress = true
 		}
 
+		req.SetOption(progressOptionName, progress)
+
 		sizeFile, ok := req.Files().(files.SizeFile)
```

Comment:
Note: I don't understand what is going on here.  Why are we setting progressOptionName if it is not found.  See my note in the PostRun function.